sechub-test/src/test/java/com/daimler/sechub/test/RoutesTest.java

```diff
@@ -0,0 +1,174 @@
+package com.daimler.sechub.test;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.security.RolesAllowed;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.daimler.sechub.sharedkernel.RoleConstants;
+
+import static org.junit.Assert.*;
+
+@SuppressWarnings("rawtypes")
+public class RoutesTest {
+	
+	private Map<Class, Map<String, ArrayList<String>>> classRoutesRolesMap = new HashMap<Class, Map<String, ArrayList<String>>>();
+	
+	@Before
+	public void before() {
+		List<Class> allClasses = getAllClassesForAnnotationType(RestController.class);
+		
+		for (Class clazz : allClasses) {
+			getClassUserRolePaths(clazz);
+		}		
+	}
+	
+	@Test
+	public void routes_assigned_to_right_roles() {
+			
+		for (Entry<Class, Map<String, ArrayList<String>>> entry : classRoutesRolesMap.entrySet()) {
+			
+			String clazzName = entry.getKey().getName();
+			Map<String, ArrayList<String>> routesEntries = entry.getValue();
+			
+			for (Entry<String, ArrayList<String>> routesEntry : routesEntries.entrySet()) {
+				String route = routesEntry.getKey();
+				String[] roles = routesEntry.getValue().toArray(new String[routesEntry.getValue().size()]);
+				
+				if (!(route.contains("anonymous") || route.contains("actuator"))) {
+					assertTrue(clazzName + " with route " + route + " should contain roles.", roles.length > 0);
+				}
+				
+				if (route.contains("anonymous") || route.contains("actuator")) {
+					assertTrue(clazzName + " with route " + route + " should not contain roles.", roles.length == 0);
+				}
+				
+				if (route.contains("/api/admin")) {
+					assertTrue(clazzName + " with route " + route + " should only contain one role, ROLE_SUPERADMIN.", roles.length == 1);
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_SUPERADMIN.", Arrays.asList(roles).contains(RoleConstants.ROLE_SUPERADMIN));
+				}
+				
+				if (route.contains("/api/project") || route.contains("/api/job")) {
```

Comment:
I would like  an else if construct here

- if is anonymous: check role defined as anonymous (I think the actuator check is only relevant for future, when provinding own actuators by REST methods, but I appreciate this)
- if integration test : just ignore
- else if admin route, check only admin
- else check at least role user must be set

So this will also check future api parts that at least user access must be defined

Also: an administrator has always user role inside - so I would just check for ROLE_USER in logic in last else part